app/controllers/reports_controller.rb

```diff
@@ -12,7 +12,12 @@ def create
   end
 
   def index
-    @reports = Report.all.limit(30)
+    if params[:latitude] && params[:longitude]
+      @reports = Report
+                 .near([params[:latitude], params[:longitude]])
+    else
+      @reports = Report.all.limit(30)
+    end
```

Comment:
No. listing things near a location and listing latest location is two different things. The first could probably use a pagination when we get a lot of reports, but not a limit.